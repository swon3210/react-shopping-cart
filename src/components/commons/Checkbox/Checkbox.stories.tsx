import Checkbox, { Props } from './Checkbox';

export default {
  title: 'Components/Commons/Checkbox',
  component: Checkbox,
  argTypes: {},
};

const Template = (args: Props) => <Checkbox {...args} />;

export const Default = Template.bind({});

(Default as any).args = {
  labelText: '라벨 텍스트',
};
