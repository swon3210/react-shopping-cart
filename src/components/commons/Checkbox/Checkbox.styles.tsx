import styled from 'styled-components';
import checkSVG from '../../../assets/svgs/check.svg';

export const Checkbox = styled.div`
  display: flex;
  align-items: center;
`;

export const Input = styled.input`
  -webkit-appearance: none;
  -moz-appearance: none;
  -o-appearance: none;
  appearance: none;

  position: relative;
  width: 28px;
  height: 28px;
  border: 1px solid #22a6a2;
  border-radius: 2px;
  cursor: pointer;

  &:checked {
    background-color: #22a6a2;
    background-image: url(${checkSVG});
    background-repeat: no-repeat;
  }
`;

export const Label = styled.label`
  position: relative;
  top: -1px;
  margin-left: 12px;
  color: ${({ theme }) => theme.TEXT_COLOR};
  letter-spacing: 0.5px;
`;
