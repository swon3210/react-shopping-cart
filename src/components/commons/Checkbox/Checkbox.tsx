import * as Styled from './Checkbox.styles';

export interface Props extends Pick<React.InputHTMLAttributes<HTMLInputElement>, 'checked'> {
  labelText?: string;
  onCheck?: (isChecked: boolean) => void;
}

const Checkbox = ({ labelText, checked, onCheck }: Props) => {
  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const isChecked = event.target.checked;
    onCheck?.(isChecked);
  };
  return (
    <Styled.Checkbox>
      {labelText ? (
        <>
          <Styled.Input type="checkbox" id={labelText} checked={checked} onChange={onChange} />
          <Styled.Label htmlFor={labelText}>{labelText}</Styled.Label>
        </>
      ) : (
        <Styled.Input type="checkbox" checked={checked} onChange={onChange} />
      )}
    </Styled.Checkbox>
  );
};

export default Checkbox;
