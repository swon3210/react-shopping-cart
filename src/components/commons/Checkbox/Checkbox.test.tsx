import Checkbox from './Checkbox';
import { render } from '@testing-library/react';

describe('Checkbox Component', () => {
  it('Checkbox Snapshot', () => {
    const checkboxUtil = render(<Checkbox />);
    expect(checkboxUtil.asFragment()).toMatchSnapshot();
  });
});
