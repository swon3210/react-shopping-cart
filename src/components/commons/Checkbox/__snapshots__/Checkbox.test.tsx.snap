// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Component Checkbox Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa fcGAIO"
  >
    <input
      class="sc-gsDJrp eXOHbT"
      type="checkbox"
    />
  </div>
</DocumentFragment>
`;
